pywinauto/macos/ax_element_info.py

```diff
@@ -158,124 +174,332 @@ def _get_ax_attribute_value(self, attr):
+                    continue
+                if process and child.process_id != process:
+                    continue
+                if class_name and child.class_name != class_name:
+                    continue
+                if title and child.name != title:
+                    continue
+                if control_type and child.control_type != control_type:
+                    continue
+                if subrole and child.subrole != subrole:
+                    continue
+                filtred_res.append(child)
+            return filtred_res
         except AXError as exc:
-            # warnings.warn(RuntimeWarning, 'Getting AXChildren attribute caused error (code = {}): "{}"' \
-            #     ''.format(exc.err_code, exc.message))
-            print('Getting AXChildren attribute caused error (code = {}): "{}"' \
-                ''.format(exc.err_code, exc.message))
-            print(type(self.ref))
             return []
 
-    def descendants(self):
-        all_desc = []
+    def descendants(self, **kwargs):
+        process = kwargs.get("process", None)
+        class_name = kwargs.get("class_name", None)
+        title = kwargs.get("title", None)
+        control_type = kwargs.get("control_type", None)
+        depth = kwargs.get("depth", None)
+
         liste = self.children()
-        def _collect_desc(elem):
-            children= elem.children()
-            if children is not None:
+        all_desc = list(liste)
+        def _collect_desc(elem,lvl):
+            next_lvl = lvl+1
+            children = elem.children()
+            if children is not None and len(children) > 0:
                 for child in children:
                     all_desc.append(child)
-                    _collect_desc(child)
+                    if depth and next_lvl > depth:
+                        continue
+                    else:
+                        _collect_desc(child,next_lvl)
         for child in liste:
-            _collect_desc(child)
-        return all_desc
+            _collect_desc(child,1)
+        filtred_res = []
+        for desc in all_desc:
+            if process and desc.process_id != process:
+                continue
+            if class_name and desc.class_name != class_name:
+                continue
+            if title and desc.name != title:
+                continue
+            if control_type and desc.control_type != control_type:
+                continue
+            filtred_res.append(desc)
+        return filtred_res
 
     @property
     def name(self):
-        if self.ref is None:
+        if self.is_desktop:
             return "Desktop"
-        try:
-            return self._get_ax_attribute_value("AXTitle")
-        except Exception:
+        if self.control_type == "MenuBar":
+            return "MenuBar"
+        name_related_attrs_list = [
+            "AXTitle",
+            "AXLabel",
+            "AXValue",
+            "AXIdentifier",
+            "AXDescription",
```

Comment:
Maybe split these attributes into separate properties. Or just don't use "AXDescription" as it is already splitted.